tests/conftest.py

```diff
@@ -0,0 +1,30 @@
+# Sharing fixtures
+# Ref: https://docs.pytest.org/en/6.2.x/fixture.html#scope-sharing-fixtures-across-classes-modules-packages-or-session
+
+import os
+import subprocess
+import shlex
+import shutil
+import tempfile
+import pytest
+
+@pytest.fixture(scope="module")
+def git_repo():
+    git_extras_cwd = os.getcwd()
+    tmp_dir = tempfile.mkdtemp()
+    tmp_file_a = tempfile.mkstemp(dir=tmp_dir)
+    tmp_file_b = tempfile.mkstemp(dir=tmp_dir)
+    os.chdir(tmp_dir)
+    result = subprocess.run(shlex.split("git init"), capture_output=True)
```

Comment:
Do you think it would be difficult to know what command is really being ran if using an external library? There could be a mismatch between the test suite and what users would usually run on the command line.

Reply:
It provides a simple DSL to run git command as in the shell:
https://gitpython.readthedocs.io/en/stable/tutorial.html#using-git-directly

This way is readable.

Reply:
Oh I see, thanks! That's quite readable